packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.ts

```diff
@@ -0,0 +1,112 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+/// !cdk-integ PipelineStack pragma:set-context:@aws-cdk/core:newStyleStackSynthesis=true
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import {
+  App,
+  Stack,
+  StackProps,
+  Stage,
+  StageProps,
+  RemovalPolicy,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import * as pipelines from 'aws-cdk-lib/pipelines';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+
+const regionalBuckets: {[key: string]: string} = {
+  'us-east-1': 'us-east-1-newpipeline-with-cross-region-replication-buckets',
+  'us-west-2': 'us-west-2-newpipeline-with-cross-region-replication-buckets',
+};
+
+interface CrossRegionReplicationBuckets {
+  [key: string]: s3.Bucket;
+}
+
+class RegionalS3Stack extends Stack {
+  public bucket: s3.Bucket;
+
+  constructor(scope: Construct, id: string, props?: StackProps, bucketName?: string) {
+    super(scope, id, props);
+
+    this.bucket = new s3.Bucket(this, 'RegionalBucket', {
+      bucketName: bucketName,
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      enforceSSL: true,
+      versioned: true,
+      autoDeleteObjects: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+}
+
+class PipelineStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps, crossRegionReplicationBuckets?: CrossRegionReplicationBuckets) {
+    super(scope, id, props);
+
+    const pipeline = new pipelines.CodePipeline(this, 'Pipeline', {
+      synth: new pipelines.ShellStep('Synth', {
+        input: pipelines.CodePipelineSource.gitHub(
+          'jose-clickup/cdk-pipelines-demo',
```

Comment:
Hi @jose-clickup thanks for the clarifications. I am not comfortable with using a source repository that is not owned by us in this integ test. Can you use the repo in `aws-samples` instead? Or are there some permissioning issues beneath the surface here as well. Sorry, I've forgotten how much of a pain pipelines integ tests are.

Reply:
@kaizencc there are numerous examples of this pattern throughout the code.

```
❯ rg cdk-pipelines-demo

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.js.snapshot/PipelineStackPipeline9DB740AF.dot
10:"Source.rix0rrr/cdk-pipelines-demo" -> "Build.Synth";
189:"Source.rix0rrr/cdk-pipelines-demo";
190:"BEGIN Source" -> "Source.rix0rrr/cdk-pipelines-demo";
191:"Source.rix0rrr/cdk-pipelines-demo" -> "END Source";

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.js.snapshot/tree.json
343:                                "name": "rix0rrr_cdk-pipelines-demo",
357:                                  "Repo": "cdk-pipelines-demo",
1953:                      "rix0rrr_cdk-pipelines-demo": {
1954:                        "id": "rix0rrr_cdk-pipelines-demo",
1955:                        "path": "PipelineStack/Pipeline/Pipeline/Source/rix0rrr_cdk-pipelines-demo",
1959:                            "path": "PipelineStack/Pipeline/Pipeline/Source/rix0rrr_cdk-pipelines-demo/WebhookResource",
1973:                                "targetAction": "rix0rrr_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.js.snapshot/manifest.json
115:        "/PipelineStack/Pipeline/Pipeline/Source/rix0rrr_cdk-pipelines-demo/WebhookResource": [

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.js.snapshot/PipelineStack.template.json
253:         "Repo": "cdk-pipelines-demo",
258:        "Name": "rix0rrr_cdk-pipelines-demo",
1864:    "TargetAction": "rix0rrr_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline.ts
14:        input: pipelines.CodePipelineSource.gitHub('rix0rrr/cdk-pipelines-demo', 'main'),

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.ts
16:        input: pipelines.CodePipelineSource.gitHub('tkglaser/cdk-pipelines-demo', 'main'),

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.ts
51:          'jose-clickup/cdk-pipelines-demo',

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.js.snapshot/PipelineStack.template.json
161:         "Repo": "cdk-pipelines-demo",
166:        "Name": "jose-clickup_cdk-pipelines-demo",
468:    "TargetAction": "jose-clickup_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.js.snapshot/PipelineStackPipeline9DB740AF.dot
10:"Source.jose-clickup/cdk-pipelines-demo" -> "Build.Synth";
52:"Source.jose-clickup/cdk-pipelines-demo";
53:"BEGIN Source" -> "Source.jose-clickup/cdk-pipelines-demo";
54:"Source.jose-clickup/cdk-pipelines-demo" -> "END Source";

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.js.snapshot/manifest.json
239:        "/PipelineStack/Pipeline/Pipeline/Source/jose-clickup_cdk-pipelines-demo/WebhookResource": [

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-region-replication-buckets.js.snapshot/tree.json
807:                                "name": "jose-clickup_cdk-pipelines-demo",
821:                                  "Repo": "cdk-pipelines-demo",
1112:                      "jose-clickup_cdk-pipelines-demo": {
1113:                        "id": "jose-clickup_cdk-pipelines-demo",
1114:                        "path": "PipelineStack/Pipeline/Pipeline/Source/jose-clickup_cdk-pipelines-demo",
1118:                            "path": "PipelineStack/Pipeline/Pipeline/Source/jose-clickup_cdk-pipelines-demo/WebhookResource",
1133:                                "targetAction": "jose-clickup_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.js.snapshot/manifest.json
127:        "/PipelineStack/Pipeline/Pipeline/Source/tkglaser_cdk-pipelines-demo/WebhookResource": [

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.js.snapshot/PipelineStack.template.json
358:         "Repo": "cdk-pipelines-demo",
363:        "Name": "tkglaser_cdk-pipelines-demo",
1978:    "TargetAction": "tkglaser_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.js.snapshot/tree.json
480:                                "name": "tkglaser_cdk-pipelines-demo",
494:                                  "Repo": "cdk-pipelines-demo",
2099:                      "tkglaser_cdk-pipelines-demo": {
2100:                        "id": "tkglaser_cdk-pipelines-demo",
2101:                        "path": "PipelineStack/Pipeline/Pipeline/Source/tkglaser_cdk-pipelines-demo",
2105:                            "path": "PipelineStack/Pipeline/Pipeline/Source/tkglaser_cdk-pipelines-demo/WebhookResource",
2119:                                "targetAction": "tkglaser_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-cross-account-keys.js.snapshot/PipelineStackPipeline9DB740AF.dot
10:"Source.tkglaser/cdk-pipelines-demo" -> "Build.Synth";
189:"Source.tkglaser/cdk-pipelines-demo";
190:"BEGIN Source" -> "Source.tkglaser/cdk-pipelines-demo";
191:"Source.tkglaser/cdk-pipelines-demo" -> "END Source";

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.js.snapshot/PipelineStackPipeline9DB740AF.dot
10:"Source.colifran/cdk-pipelines-demo" -> "Build.Synth";
189:"Source.colifran/cdk-pipelines-demo";
190:"BEGIN Source" -> "Source.colifran/cdk-pipelines-demo";
191:"Source.colifran/cdk-pipelines-demo" -> "END Source";

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.js.snapshot/tree.json
359:                                "name": "colifran_cdk-pipelines-demo",
373:                                  "Repo": "cdk-pipelines-demo",
1969:                      "colifran_cdk-pipelines-demo": {
1970:                        "id": "colifran_cdk-pipelines-demo",
1971:                        "path": "PipelineStack/Pipeline/Pipeline/Source/colifran_cdk-pipelines-demo",
1975:                            "path": "PipelineStack/Pipeline/Pipeline/Source/colifran_cdk-pipelines-demo/WebhookResource",
1989:                                "targetAction": "colifran_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.js.snapshot/manifest.json
115:        "/PipelineStack/Pipeline/Pipeline/Source/colifran_cdk-pipelines-demo/WebhookResource": [

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.js.snapshot/PipelineStack.template.json
253:         "Repo": "cdk-pipelines-demo",
258:        "Name": "colifran_cdk-pipelines-demo",
1864:    "TargetAction": "colifran_cdk-pipelines-demo",

packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.newpipeline-with-codebuild-logging.ts
20:        input: pipelines.CodePipelineSource.gitHub('colifran/cdk-pipelines-demo', 'main'),
```

This is pretty urgent for us. Can we get this merged and then, if you guys feel strongly about it, can you please do a cleanup?

Reply:
@kaizencc, there are permissions issues because we're not part of the `aws-samples` org (despite it being public). This makes sense as from CodePipeline you are only able to select a repo that you own (listed using your github-token) from the GitHub connection